packages/cli/test/integration.js

```diff
@@ -1087,29 +1082,48 @@ test('Deploy `api-env` fixture and test `vercel env` command', async t => {
     t.is(exitCode, 0, formatOutput({ stderr, stdout }));
   }
 
-  await vcLink();
-  await vcEnvLsIsEmpty();
-  await vcEnvAddWithPrompts();
-  await vcEnvAddFromStdin();
-  await vcEnvAddFromStdinPreview();
-  await vcEnvAddFromStdinPreviewWithBranch();
-  await vcEnvLsIncludesVar();
-  await createEnvWithDecryptableSecret();
-  await vcEnvPull();
-  await vcEnvPullOverwrite();
-  await vcEnvPullConfirm();
-  await vcDeployWithVar();
-  await vcDevWithEnv();
-  fs.unlinkSync(path.join(target, '.env'));
-  await vcDevAndFetchCloudVars();
-  await enableAutoExposeSystemEnvs();
-  await vcEnvPullFetchSystemVars();
-  fs.unlinkSync(path.join(target, '.env'));
-  await vcDevAndFetchSystemVars();
-  await vcEnvRemove();
-  await vcEnvRemoveWithArgs();
-  await vcEnvRemoveWithNameOnly();
-  await vcEnvLsIsEmpty();
+  function vcEnvRemoveByName(name) {
+    return execa(binaryPath, ['env', 'rm', name, '-y', ...defaultArgs], {
+      reject: false,
+      cwd: target,
+    });
+  }
+
+  async function vcEnvRemoveAll() {
+    await vcEnvRemoveByName('MY_PREVIEW');
+    await vcEnvRemoveByName('MY_STDIN_VAR');
+    await vcEnvRemoveByName('MY_DECRYPTABLE_SECRET_ENV');
+    await vcEnvRemoveByName('MY_NEW_ENV_VAR');
+  }
+
+  try {
+    await vcEnvRemoveAll();
+    await vcLink();
+    await vcEnvLsIsEmpty();
+    await vcEnvAddWithPrompts();
+    await vcEnvAddFromStdin();
+    await vcEnvAddFromStdinPreview();
+    await vcEnvAddFromStdinPreviewWithBranch();
+    await vcEnvLsIncludesVar();
+    await createEnvWithDecryptableSecret();
+    await vcEnvPull();
+    await vcEnvPullOverwrite();
+    await vcEnvPullConfirm();
+    await vcDeployWithVar();
+    await vcDevWithEnv();
+    fs.unlinkSync(path.join(target, '.env'));
+    await vcDevAndFetchCloudVars();
+    await enableAutoExposeSystemEnvs();
+    await vcEnvPullFetchSystemVars();
+    fs.unlinkSync(path.join(target, '.env'));
+    await vcDevAndFetchSystemVars();
+    await vcEnvRemove();
+    await vcEnvRemoveWithArgs();
+    await vcEnvRemoveWithNameOnly();
+    await vcEnvLsIsEmpty();
+  } finally {
+    await vcEnvRemoveAll();
+  }
```

Comment:
This test was failing, but failures polluted the env variables in a way that caused future test suite runs to fail. This change makes sure the env variables are always deleted after the test runs (failure or not).